packages/calling/src/CallHistory/CallHistory.ts

```diff
@@ -128,6 +143,43 @@ export class CallHistory extends Eventing<CallHistoryEventTypes> implements ICal
           );
         }
       }
+      // Check the calling backend
+      const callingBackend = getCallingBackEnd(this.webex);
+      if (callingBackend === CALLING_BACKEND.UCM) {
+        // Check if userSessions exist and the length is greater than 0
+        if (this.userSessions[USER_SESSIONS] && this.userSessions?.[USER_SESSIONS]?.length > 0) {
+          // Check if cucmDN exists and is valid in any of the userSessions
+          const hasCucmDN = this.userSessions[USER_SESSIONS].some(
+            (session: UserSession) => session.self.cucmDN && session.self.cucmDN.length > 0
+          );
+          // If any user session has cucmDN, proceed to fetch line data
+          if (hasCucmDN) {
+            // Fetch the Lines data
+            const ucmLinesResponse = await this.fetchUCMLinesData();
+
+            // Check if the Lines API response was successful
+            if (ucmLinesResponse.statusCode === 200 && ucmLinesResponse?.data?.lines?.devices) {
```

Comment:
Are the ? necessary for all the members in `ucmLinesResponse?.data?.lines?.devices`. Isn't ucmLinesResponse always going to have a value?

Reply:
Please add fixtures and test for all the '?' variables. For every ? entry, there needs to be a test that doesn't have that element in the json.

Reply:
We have removed the question marks from ucmLinesResponse.data as suggested, but we have kept the question mark at lines (ucmLinesResponse.data.lines?.devices) since it is an optional parameter. Additionally, we have written test cases to cover scenarios where line devices may not be present in the response.

Reply:
The test should cover that lines isn't present in the response right? The question mark indicates that the lines object is empty? devices being empty should have a different condition. If lines is always going to be present, then we should not make it an optional object in the `UCMLinesResponse` type.